pkg/plugins/gateway/algorithms/prefix_cache.go

```diff
@@ -18,113 +18,205 @@ package routingalgorithms
 
 import (
 	"fmt"
+	"math"
 	"math/rand"
+	"sort"
 	"strconv"
 
-	"github.com/vllm-project/aibrix/pkg/plugins/gateway/prefixcacheindexer"
-	"github.com/vllm-project/aibrix/pkg/plugins/gateway/prefixcacheindexer/tokenizer"
+	"github.com/vllm-project/aibrix/pkg/cache"
+	"github.com/vllm-project/aibrix/pkg/plugins/gateway/algorithms/prefixcacheindexer"
+	"github.com/vllm-project/aibrix/pkg/plugins/gateway/algorithms/tokenizer"
 	"github.com/vllm-project/aibrix/pkg/types"
 	"github.com/vllm-project/aibrix/pkg/utils"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/klog/v2"
 )
 
-var (
-	RouterPrefixCache types.RoutingAlgorithm = "prefix-cache"
-)
-
-func init() {
-	RegisterDelayedConstructor(RouterPrefixCache, NewPrefixCacheRouter)
-}
-
 const (
-	defaultPrefixCacheMatchThresholdPercent = 50
+	defaultPodRunningRequestImbalanceAbsCount = 8
 )
 
 var (
-	prefixCacheMatchThresholdPercent = getPrefixCacheMatchThresholdPercent()
+	RouterPrefixCache                  types.RoutingAlgorithm = "prefix-cache"
+	podRunningRequestImbalanceAbsCount int                    = getPodRequestImbalanceAbsCount()
 )
 
-func getPrefixCacheMatchThresholdPercent() int {
-	value := utils.LoadEnv("AIBRIX_PREFIX_CACHE_MATCH_THRESHOLD_PERCENT", "")
+func getPodRequestImbalanceAbsCount() int {
+	value := utils.LoadEnv("AIBRIX_PREFIX_CACHE_POD_RUNNING_REQUEST_IMBALANCE_ABS_COUNT", "")
```

Comment:
Let's update the docs for new added envs

Reply:
Different GPU has different compute capacity, using absolute numbers looks like not that scalable. How can we generate this number?

Reply:
We can leverage GPU optimizer to generate recommendations for abs numbers to define imbalance.